sorting/wave_sort.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file
+ * @brief Implementation of [Wave sort] algorithm
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation.